scripts/RT/magic.R

```diff
@@ -166,26 +218,215 @@ if (SEag < 0.1){
   SEag <- 7.0
 }
   
-#Slagm = Aggregate Geometric Mean Diameter [SWEEP document page 17]
-Slagm <- exp(1.343-2.235*Sandpct-1.226*Siltpct-0.0238*(Sandpct/Claypct)^3 
-             + 33.6*Organicpct+6.85*CaCO3pct)*(1+0.006*Depthmm)
-#adjustments for absolute range allowed in SWEEP document page 17
-if (Slagm < 0.03){
-  Slagm <- 0.03
-} else if (Slagm > 30.0){
-  Slagm <- 30.0
-}
-
-
-#S0ags = Aggregate Geometric Standard Deviation (that's a zero "0", not an "O") [SWEEP document page 17]
-S0ags <- 1.0 / (0.0203 + 0.00193*Slagm + 0.074 / Slagm^0.5)
-#adjustments for absolute range allowed in SWEEP document page 17
-if (S0ags < 1.0){
-  S0ags <- 1.0
-} else if (S0ags > 20.0){
-  S0ags <- 20.0
-}
-
+#Obsolete as of 5/3/24 (using WEPS-based aggregate equations instead)
+  #Slagm = Aggregate Geometric Mean Diameter [SWEEP document page 17]
+  #Slagm <- exp(1.343-2.235*Sandpct-1.226*Siltpct-0.0238*(Sandpct/Claypct)^3 
+  #           + 33.6*Organicpct+6.85*CaCO3pct)*(1+0.006*Depthmm)
+  #adjustments for absolute range allowed in SWEEP document page 17
+  #if (Slagm < 0.03){
+  # Slagm <- 0.03
+#  } else if (Slagm > 30.0){
+#  Slagm <- 30.0
+#}
+
+  #S0ags = Aggregate Geometric Standard Deviation (that's a zero "0", not an "O") [SWEEP document page 17]
+  #S0ags <- 1.0 / (0.0203 + 0.00193*Slagm + 0.074 / Slagm^0.5)
+  #adjustments for absolute range allowed in SWEEP document page 17
+  #if (S0ags < 1.0){
+    #S0ags <- 1.0
+  #} else if (S0ags > 20.0){
+    #S0ags <- 20.0
+  #}
+
+#*****NEW AGGREGATE CALCULATIONS (From "WEPS_test2")
+  #*****Management Decisions***** I believe matches the WEPS calculations (5/3/24)
+  #*This currently only works for corn; need some kind of flag to determine what crop is growing?
+  management <- WEPProt$mgmt_flag[i]
+  operation<-WEPProt$operation[i]
+  details1<-WEPProt$details1[i]
+  
+  if (management==1){
+    #adjust nondimensional gmd based on the management occuring
+    x<-log(cslagm_0)
+    #chisel plow
+    #if (operation<-"OpCropDef.CHISSTSP"){
+    #y<-0.6924*x+0.1634
+    if (operation=="OpCropDef.FCSTACDP") {#spring disk (avg corn and soy)
+      y<-0.688*x-0.00665
+      
+    } else if (operation=="OpCropDef.PLDDO") { #field finisher (no field finisher for WEPS, assuming its the same as cultivate)
+      y<-0.8097*x-0.11675
+      
+      #} else if (op2=="CropDef.Cor_0967"){ #plant corn
+      #y<-0.8752*x-0.3286
+      
+      #}else if (op2=="CropDef.Soy_2194"){ #plant soybean
+      #y<-0.8293*x-0.2664
+    } else if (operation=="OpCropDef.CHISSTSP" && details1=="{0.203200, 2}") { #soybean fall chisel
+      y<-0.7768*x-0.3976
+    }else if (operation=="OpCropDef.CHISSTSP" && details1=="{0.203200, 1}"){ #corn fall chisel
+      y<-0.6924*x+0.1634
+    }else
+      y<-log(cslagm_0)
+    
+    
+    cslagm_0<-exp(y)
+    gmd0<-(cslagm_0-cslmin)/(cslmax-cslmin)
+    
+    #management not occuring, nondimensional gmd does not need to be adjusted  
+  }else{ 
+    gmd0<-gmd0

```

Comment:
Can this else conditional be omitted? I don't think it does anyway.

Reply:
I think that it can be omitted. In some quick tests, it doesn't seem like it is necessary. Though I am nervous about getting rid of it in case there was a reason I had put it there originally